components/ListLink.styled.js

```diff
@@ -0,0 +1,31 @@
+import css from "styled-jsx/css";
+
+const { default: Link } = require("next/link");
+const { styled } = require("styled-components");
+
+const listVariants = {
+  breed: css`
+    display: flex;
+    align-items: center;
+    gap: 1rem;
+    padding: 0.5rem;
+  `,
+};
+
+const StyledLink = styled(Link)`
+  position: relative;
+  display: block;
+  padding: 1rem;
+  border-radius: 0.5rem;
+  background-color: lightgray;
+  color: black;
+  text-decoration: none;
+
+  &:hover {
+    filter: brightness(0.85);
+  }
+
+  ${({ $variant }) => listVariants[$variant] || ""};
```

Comment:
This is an interesting approach, but keep in mind that the more readable your code is, the better. I would assume that a lot of devs are not familiar with this kind of writing your variants. Your code might be more readable if you would list the styling directly underneath each other in the styled component. But still, I like it.

Reply:
thanks for the tipp
mareike mentioned variants in a earlier PR and I googled that

but I changed it to something more readable I hope